consensus/istanbul/core/commit.go

```diff
@@ -71,6 +71,9 @@ func (c *core) handleCommit(msg *message, src istanbul.Validator) error {
 	}
 
 	//logger.Error("receive handle commit","num", commit.View.Sequence)
+	if !c.valSet.CheckInSubList(msg.Hash, commit.View, src.Address()) {
+		return errNotFromCommittee
+	}
 
 	if err := c.checkMessage(msgCommit, commit.View); err != nil {
```

Comment:
Is it possible to call `CheckInSubList` inside `checkMessage`? Don't we need to check if the sender is in the committee for `preprepare` or `processBacklog`?

Reply:
We don't need to check `preprepare` message since it only has one valid sender(proposer) at a time and is checked with its own routine.  
Because `CheckInSubList` is not a common routine for all Istanbul message, I think it is better to place `CheckInSubList` in the current position.